src/ott/neural/layers.py

```diff
@@ -81,59 +78,82 @@ def __call__(self, inputs: jnp.ndarray) -> jnp.ndarray:
 
 
 class PosDefPotentials(nn.Module):
-  r"""A layer to output :math:`\frac{1}{2} ||A_i^T (x - b_i)||^2_i` potentials.
+  """A layer to output  0.5 x^T(A_i A_i^T + Diag(d_i^2))x + b_i^T x + c_i potentials.
 
   Args:
-    use_bias: whether to add a bias to the output.
-    dtype: the dtype of the computation.
-    precision: numerical precision of computation see `jax.lax.Precision`
-      for details.
-    kernel_init: initializer function for the weight matrix.
-    bias_init: initializer function for the bias.
-  """
-  dim_data: int
+  num_potentials: the dimension of the output
+  rank: the rank of the matrix used for the quadratic layer
+  use_linear: whether to add a linear layer to the output
+  use_bias: whether to add a bias to the output.
+  dtype: the dtype of the computation.
+  precision: numerical precision of computation see `jax.lax.Precision` for details.
+  kernel_quadratic_init: initializer function for the weight matrix of the quadratic layer.
+  kernel_linear_init: initializer function for the weight matrix of the linea layer.
+  bias_init: initializer function for the bias.
+  """  # noqa: E501
+
   num_potentials: int
+  rank: int = 0
```

Comment:
because, for positive `d_i`, the quadratic form will always be full rank, regardless of the rank of `A_i`, we might have to find another name.